image.go

```diff
@@ -425,16 +483,63 @@ func (b *Builder) makeContainerImageRef(manifestType string, compress archive.Co
 	if err != nil {
 		return nil, err
 	}
+	created := time.Now().UTC()
+	if historyTimestamp != nil {
+		created = historyTimestamp.UTC()
+	}
 	ref := &containerImageRef{
 		store:                 b.store,
-		container:             container,
 		compression:           compress,
 		name:                  name,
+		names:                 container.Names,
+		layerID:               container.LayerID,
+		addHistory:            true,
 		oconfig:               oconfig,
 		dconfig:               dconfig,
+		created:               created,
 		createdBy:             b.CreatedBy(),
 		annotations:           b.Annotations(),
 		preferredManifestType: manifestType,
+		exporting:             exporting,
+	}
+	return ref, nil
+}
+
+func (b *Builder) makeImageImageRef(compress archive.Compression, names []string, layerID string) (types.ImageReference, error) {
```

Comment:
This and `makeContainerImageRef` seems to share a enough code that it might be possible to extract a shared helper from the two. OTOH the code is simple enough that this might not be worth it, I have no idea.